src/scheduler/gc_work.rs

```diff
@@ -101,23 +102,26 @@ impl<VM: VMBinding> GCWork<VM> for PrepareCollector {
 /// be no other concurrent work packet that accesses plan (read or write). Otherwise, there may
 /// be a race condition.
 pub struct Release<C: GCWorkContext> {
-    pub plan: &'static C::PlanType,
+    pub plan: *const C::PlanType,
 }
 
 impl<C: GCWorkContext> Release<C> {
-    pub fn new(plan: &'static C::PlanType) -> Self {
+    pub fn new(plan: *const C::PlanType) -> Self {
         Self { plan }
     }
 }
 
+unsafe impl<C: GCWorkContext> Send for Release<C> {}
+
 impl<C: GCWorkContext + 'static> GCWork<C::VM> for Release<C> {
     fn do_work(&mut self, worker: &mut GCWorker<C::VM>, mmtk: &'static MMTK<C::VM>) {
         trace!("Release Global");
 
-        self.plan.base().gc_trigger.policy.on_gc_release(mmtk);
-
+        unsafe {
+            (*self.plan).base().gc_trigger.policy.on_gc_release(mmtk);
+        }
         // We assume this is the only running work packet that accesses plan at the point of execution
```

Comment:
We actually know this is false. See https://github.com/mmtk/mmtk-core/issues/824#issuecomment-1558661794. I think we shouldn't ignore this issue

Reply:
@qinsoon Do you have any thoughts on the above?

Reply:
I've fixed only UB problems in this PR so didn't know about this issue. So what should I do in that line? Simply remove this call?

Reply:
The discussion on this topic is not related with the PR. I make this clear so it won't confuse @playXE . I think nothing needs to be done for the PR.

I listed assumptions in https://github.com/mmtk/mmtk-core/issues/717#issuecomment-1362097538. This line is correct if the assumptions are true. The issue in #824 is a case in mark sweep that violates the assumptions: mark sweep's mutator release and plan's release both modify the block list. We either rethink our assumptions, or fix the implementation in mark sweep.